cypress/integration/IntersectionDetector/Visibility_notification/index.js

```diff
@@ -0,0 +1,43 @@
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps'
+
+Given('the detector is not visible', () => {
+    cy.visitStory('IntersectionDetector', 'Out Of View')
+    cy.window().then(win => {
+        cy.wrap(win.onVisibilityChange)
+            .as('onVisibilityChangeStub')
+            .should('be.calledOnce')
+            .should('be.calledWith', { visible: false })
+    })
+})
+
+Given('the detector is visible', () => {
+    cy.visitStory('IntersectionDetector', 'In View')
+    cy.window().then(win => {
+        cy.wrap(win.onVisibilityChange)
+            .as('onVisibilityChangeStub')
+            .should('be.calledOnce')
+            .should('be.calledWith', { visible: true })
+    })
+})
+
+When('the user scrolls the detector into the view', () => {
+    // using 311 to scroll 1px more than "just enough"
+    cy.get('[data-test="scroll-container"]').scrollTo(0, 311)
+})
+
+When('the user scrolls the detector out of the view', () => {
+    // Scroll 1px more than "just enough" to make sure
+    // the detector is completely hidden
+    cy.get('[data-test="scroll-container"]').scrollTo(0, 201)
+})
+
+Then('the callback passed to onIsVisible should be called', () => {
+    cy.get('@onVisibilityChangeStub').should(stub => {
+        expect(stub).to.be.calledTwice
+    })
+})
+
+Then(/the visible prop of its payload should be .*/, bool => {
```

Comment:
Can this be done without a regex? I don't find them the most readable personally.

Reply:
Split into two step definitions